docs/DeveloperGuide.md

```diff
@@ -320,121 +320,193 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 **MSS**
+1. User adds a service by providing details.
+1. GrAB3 adds the service to the service list.  
+1. GrAB3 displays successful additional message.
+Use case ends.
 
-0.  User adds appointment into appointment list.
-1.  GrAB3 shows the updated list of appointments.
+Extensions:  
+* 1a. Incomplete details provided.  
+  * 1a1. GrAB3 displays an error message.  
+  Use case resumes at step 1.
+        
+#### UC002: Edit a Service
+Edit the details of an existing service.
 
-    Use case ends.
+**Preconditions: Service exists in service list.**
+
+**Guarantees: Updated service list with new service details.**
+
+**MSS**
+1. User requests to list all services.
+1. GrAB3 shows a list of services.
+1. User edits an existing service by providing the details.
+1. GrAB3 update the details of the selected service.
+1. GrAB3 update the service list.
+Use case ends.
+
+Extensions:
+* 3a. Invalid index provided.
+  * 3a1. GrAB3 displays an error message.  
+  Use case resumes at step 1.
+
+* 3b. Incomplete details provided.  
+  * 3b1. GrAB3 displays an error message.  
+  Use case resumes at step 1.
+        
+#### UC003: Delete a Service
+Delete the selected service from the service list.
+
+**Preconditions: Service exists in service list.**
+
+**Guarantees: Updated service list with service specified by user deleted.**
+
+**MSS**
+1. User requests to list all services.
+1. GrAB3 shows a list of services.
+1. User request to delete a specific service.
+1. GrAB3 delete a service. 
+1. GrAB3 shows the updated list view.
+Use case ends.
 
 **Extensions**
+* 3a. Invalid index provided.
+  * 3a1. GrAB3 displays an error message.  
+  Use case resumes at step 1.
 
-* 2a. The specified module code is invalid.
+#### UC004: Find a Service
+Find a service in the service list.
 
-    * 2a1. GrAB3 shows an error message.
+**Preconditions: Service exists in service list.**
 
-    Use case resumes at step 1.
+**Guarantees: Display all services in the service list that matches the search value.**
+
+**MSS**
+1. User requests to list all services that matches the search value.
+1. GrAB3 searches for services that matches search value.
+1. GrAB3 show list of all services that matches the search value.
+Use case ends.
+
+**Extensions**
+* 2a. No service matches the search value.
+  * 2a1. GrAB3 displays empty list.
+  Use case resumes at step 1.
+  
+#### UC005: List Services 
+List all the services in the service list.
+
+**Preconditions: Service exists in service list.**
+ 
+**Guarantees: All services in the service list will be displayed.**
+ 
+**MSS**
+1. User requests to list all services.
+1. GrAB3 display all services in the service list.
+Use case ends.
+
+**Extensions**
+* 1a. No service in service list.
+  * 1a1. GrAB3 displays empty list.
+  Use case resumes at step 1.
+  
+#### UC005: Clear Services
+Clear all services in the service list.
+
+**Preconditions: Service exists in service list**
+
+**Guarantees: An empty revenue list.**
```

Comment:
Does it guarantee an empty service list as well?

Reply:
What you mean? If clear than should be empty?